README.md

```diff
@@ -1,84 +1,343 @@
+### npm local packages
 
-    class { 'nodejs':
-      node_pkg    => 'nodejs010',
-      npm_pkg     => 'nodejs010-npm',
-      dev_pkg     => 'nodejs010-devel',
-      dev_package => true,
-    }
+nodejs::npm is used for the local installation of npm packages:
 
-Debian users may want to use apt::pin to pin package installation priority on squeeze. See [puppetlabs-apt](https://github.com/puppetlabs/puppetlabs-apt) for more information.
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present',
+}
+```
 
-    apt::pin { 'sid': priority => 100 }
+Installing a specific version:
 
-### npm package
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => '2.5.9',
+}
+```
 
-Two types of npm packages are supported.
+Installing the latest version:
 
-* npm global packages are supported via ruby provider for puppet package type.
-* npm local packages are supported via puppet define type nodejs::npm.
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'latest',
+}
+```
 
-For more information regarding global vs. local installation see [nodejs blog](http://blog.nodejs.org/2011/03/23/npm-1-0-global-vs-local-installation/)
+Installing from GitHub:
 
-### package
-npm package provider is an extension of puppet package type which supports versionable and upgradeable. The package provider only handles global installation:
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present'
+  source  => 'strongloop/express',
+}
+```
 
-Example:
+Installing from a git repository:
 
-    package { 'express':
-      ensure   => present,
-      provider => 'npm',
-    }
-    
-    package { 'mime':
-      ensure   => '1.2.4',
-      provider => 'npm',
-    }
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present'
+  source  => 'git+https://git@github.com/strongloop/expressjs.com.git',
+}
+```
 
-### nodejs::npm
-nodejs::npm is suitable for local installation of npm packages:
+Installing from a remote tarball:
 
-    nodejs::npm { '/opt/razor:express':
-      ensure  => present,
-      version => '2.5.9',
-    }
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present'
+  source  => 'https://server.domain/express.tgz',
+}
+```
 
-nodejs::npm title consists of filepath and package name seperate via ':', and support the following attributes:
+Installing tagged packages:
 
-* ensure: present, absent.
-* version: package version (optional).
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present'
+  tag     => 'tagname',
+}
+```
+
+Installing scoped packages:
+
+```puppet
+nodejs::npm { '@scope_name/express|/opt/packages':
+  ensure  => 'present'
+}
+```
+
+Installing packages from a local tarball:
+
+```puppet
+nodejs::npm { 'express|/opt/packages':
+  ensure  => 'present'
+  source  => '/local/repository/npm_packages/express.tgz'
+}
+```
+
+The nodejs::npm title consists of a package name separated by a '|', and
```

Comment:
What about the '@scope_name' part? I think it would make sense to describe this before the examples.